learners/BF_measles.Rmd

```diff
@@ -0,0 +1,884 @@
+
+# Print the contact matrix
+#contact_matrix
+
+# Define the population structure for Burkina Faso
+initial_i <- 1 / bf_pop
+initial_conditions <- c(S = 1 - initial_i, E = 0, I = initial_i, R = 0, V = 0)
+
+# Replicate initial conditions for all age groups
+initial_conditions_matrix <- rbind(
+  initial_conditions,
+  initial_conditions,
+  initial_conditions,
+  initial_conditions,
+  initial_conditions
+)
+
+rownames(initial_conditions_matrix) <- rownames(contact_matrix)
+
+# Print the initial conditions
+#initial_conditions_matrix
+
+# Create the population object
+bf_population <- epidemics::population(
+  name = "Burkina Faso",
+  contact_matrix = contact_matrix,
+  demography_vector = demography_vector,
+  initial_conditions = initial_conditions_matrix
+)
+
+# Print the population object
+# bf_population
+```
+
+```{r}
+# Run the epidemic model
+output <- epidemics::model_default(
+  population = bf_population,
+  transmission_rate = 15 / 5,  # R0/infectious period = 15/5 = 3
+  infectiousness_rate = 1 / 8,  # 1/latent period (pre-infectious phase)
+  recovery_rate = 1 / 5,  # 1/infectious period
+  time_end = 122,  # Simulation over 122 days
+  increment = 1  # Daily increments
+)
+
+# Plot the epidemic curve
+output %>%
+  filter(compartment %in% c("exposed", "infectious")) %>%
+  ggplot(aes(
+    x = time,
+    y = value,
+    col = demography_group,
+    linetype = compartment
+  )) +
+  geom_line(linewidth = 1.2) +
+  scale_y_continuous(labels = scales::comma) +
+  scale_colour_brewer(palette = "Dark2", name = "Age group") +
+  expand_limits(y = c(0, 500e3)) +
+  coord_cartesian(expand = FALSE) +
+  theme_bw() +
+  theme(
+    legend.position = "top",
+    legend.text = element_text(size = 10, hjust = 0, lineheight = 121)
+  ) +
+  labs(
+    x = "Simulation time (days)",
+    linetype = "Compartment",
+    y = "Individuals",
+    caption =
+      expression(
+        paste(
+          "SIER model showing the trajectories of the number of ",
+          "exposed and infected individuals ",
+          italic(R)[0], " = 15 for different age groups."
+        )
+      )
+  ) +
+  guides(
+    color = guide_legend(nrow = 2)
+  )
+```
+
+
+## Effective Reproduction Number $\mathcal{R}_{eff}$
+
+Age-structured models account for heterogeneous characteristics in the
+population, such as:
+
+-   Demographic differences in social contacts,
+-   Variations in demographic distribution, and
+-   Differences in susceptibility to infection.
+
+These factors result in different $\mathcal{R}_0$ values for each age
+group. The **effective reproduction number** $\mathcal{R}_{eff}$
+represents the overall reproduction number across all age groups while
+accounting for these differences. It is calculated using the following
+formula:
+
+[$$
+\begin{equation}
+\mathcal{R}_{eff} = \rho(C) \cdot \lambda \cdot \gamma 
+\end{equation}
+ \qquad(5)$$]{#eq-reff}
+
+Where:
+
+-   $\rho(C)$ is the **largest eigenvalue** of the contact matrix $C$,
```

Comment:
This matrix is already normalised, right? So if R0=1, it would produce R_eff=1?

Reply:
Yes, the martix is normalize but for R0 =1, gives R_eff = 0.56...!